README.md

```diff
@@ -1,32 +1,65 @@
+
+## Testing Plan
+Currently, we only provide a single test case for our checkpointing approach. It involves creating a data table, performing some basic transactions, taking checkpoints and recovering from the checkpoint. We eventually compare the recovered table with the original table to examine the correctness of our implementation.
+
+In the final stage, the recovery framework should work for all existing test cases in recovery_test.cpp, with a checkpoint taken before shutting down the system and previous log deleted.
+
+We will eventually rely on recovery_benchmark.cpp to compare the performance of our approach to the existing one.
+
+
+## Trade-offs and Potential Problems
+For the checkpoint part, the current design creates several threads and lets one thread write the data for exactly one table to disk. To inform the recovery stage the corresponding table for each file, the filename format db_oid-table_oid is used. Considering catalog table recovery is a separated process from current user table recovery, the order of the tables is not recorded. This design simplifies the implementation of multithreading in this phase, but may introduce two potential problems. First, the catalog table may need to be recovered from the checkpoint in some situations. The database has to recover the catalog table before recovering any other tables. This requires the checkpoints to record the order of table data files. Also, if the order of files is recorded in the filename or in any other data structures, traversing all the files to find the one for the catalog table increases the total recovery time when doing the recovery. 
+
+Current recovery mechanism recovers data tables directly from the checkpoints without touching the sql tables related to them. Also, the checkpoint class accesses the data table directly from a given sql table. It is possible that later, one sql table can have multiple data tables related to it. Thus, the checkpoint and recovery mechanism may need to be modified to handle the recovery of sql tables as well in the future. 
+
+Finally, current checkpoints record the content field of each block in a data table for the convenience of software engineering, which includes repeated block headers. If given many giant tables, the redundant headers might consume relatively large space on the disk. 
+
+
+## Future Work
+As mentioned in the previous section, the current design of checkpoints taking creates one file for each table with the filename db_oid-table_oid. Future work can modify the write_to_disk function in current checkpoint class to record all the tables in one large file, which ensures the order correctness of the tables to be recovered. Also, in the current design of checkpoints taking, each table is processed by one thread. People can consider a better multithreading plan such as allowing threads which finish early to process a part of work from those process large tables to reduce the total time for taking checkpoints. 
```

Comment:
Have added comments with some more suggestions along the lines of what @pdaviso mentioned above for the checkpoint.cpp file. 

Also, this design doc needs to be updated because I don't think a thread handles a single table as of now.